aptos-move/framework/src/natives/cryptography/bls12381.rs

```diff
@@ -636,12 +640,58 @@ pub fn native_bls12381_verify_signature_share(
     gas_params.bls12381_verify_signature_helper(_context, _ty_args, arguments, check_pk_subgroup)
 }
 
+#[cfg(feature = "testing")]
+pub fn native_generate_keys(
+    _context: &mut NativeContext,
+    _ty_args: Vec<Type>,
+    mut _arguments: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    let key_pair = KeyPair::<PrivateKey, PublicKey>::generate(&mut OsRng);
+    Ok(NativeResult::ok(
+        InternalGas::zero(),
+        smallvec![
+            Value::vector_u8(key_pair.private_key.to_bytes()),
+            Value::vector_u8(key_pair.public_key.to_bytes()),
+        ],
+    ))
+}
+
+pub fn native_sign(
+    _context: &mut NativeContext,
+    _ty_args: Vec<Type>,
+    mut arguments: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    let msg = pop_arg!(arguments, Vec<u8>);
+    let sk_bytes = pop_arg!(arguments, Vec<u8>);
+    let sk = PrivateKey::try_from(sk_bytes.as_slice()).unwrap();
+    let sig = sk.sign_arbitrary_message(msg.as_slice());
+    Ok(NativeResult::ok(
+        InternalGas::zero(),
+        smallvec![Value::vector_u8(sig.to_bytes()),],
+    ))
+}
+
+pub fn native_aggr_sign(
```

Comment:
Gas parameters?

Reply:
I'm also not seeing this native being included in the native list.

Reply:
This is a test-only native, so no gas parameters!